app/controllers/sessions_controller.rb

```diff
@@ -1,2 +1,27 @@
 class SessionsController < ApplicationController
+  before_action :require_login, except: [:new, :create]
+  skip_before_filter :verify_authenticity_token, only: [:create]
+
+  def new
+  end
+
+  def create
+   auth_hash = request.env['omniauth.auth']
+   if auth_hash["uid"]
+     @user = User.find_or_create_from_omniauth(auth_hash)
+     if @user
+       session[:user_id] = @user.id
+       redirect_to feed_path, notice: "Welcome to CurateTV. Happy Curating!"
+     else
+       redirect_to root_path, notice: "Failed to save the user."
+     end
+   else
+     redirect_to root_path, notice: "Failed to authenticate."
+   end
```

Comment:
Avoid nesting conditionals! With this one, you might be able to set `@user` before the conditional block, then say `if auth_hash["uid"] && @user` to avoid one level of nesting.